### ISSUE TITLE
Support set operator in filter argument

### ISSUE THREAD

**Issue by jenrik** [neticdk-k8s/ic#6]
It's currently not possible to have multiple filter conditions on the same field. Calling `ic get clusters --filter clusterID!=aaaa --filter clusterID!=bbbb` will only use the last filter for a given field.

Adding support for set-based selectors like the ones kubectl support would allow for "clusterID notin (aaaa, bbbb)" which would solve some of the issue.

https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement

**Comment by KimNorgaard** [comment]
I looked into this. It's a little hard or time consuming to implement because the search engine and parsing uses a map for the fields (so currently it doesn't support multiple conditions per field - last man wins). I'll try to see if it's viable to change it.

**Comment by KimNorgaard** [comment]
Well well well. Turns out I already had thought of future me and implemented NotIn and In in the backend, so I just needed a few tweeks here and there.

So now, this is supported:

```bash
ic get clusters -f 'clusterID in (clusterA.providerA, clusterB.providerB)'
ic get clusters -f 'clusterID notin (clusterA.providerA, clusterB.providerB)'
```

The parentheses are optional as they are simply `Trim()`'d from the search value. Same goes for spaces before or after commas.

The behavior of adding multiple filters for the same field still results only in one filter being applied. But I guess this solved your specific problem?

**Comment by KimNorgaard** [comment]
Fixed with 82e53d1198d41c086824c6eb74fb9a3a6d640592
